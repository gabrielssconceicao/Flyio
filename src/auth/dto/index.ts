import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
