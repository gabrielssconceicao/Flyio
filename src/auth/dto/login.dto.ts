import { IsNotEmpty, IsString } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';
export class LoginDto {
  @IsNotEmpty({
    message: dtoErrorMessages('Username or email', DtoErrorType.REQUIRED),
  })
  @IsString()
  usernameOrEmail: string;

  @IsNotEmpty({ message: dtoErrorMessages('Password', DtoErrorType.REQUIRED) })
  @IsString()
  password: string;
}
