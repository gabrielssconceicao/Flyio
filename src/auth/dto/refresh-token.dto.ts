import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString } from 'class-validator';

export class RefreshTokenDto {
  @ApiProperty({
    example: 'krcio48hgGYIFYJfuyfJYAYu',
    description: 'Refresh token',
  })
  @IsString()
  @IsNotEmpty()
  refreshToken: string;
}
