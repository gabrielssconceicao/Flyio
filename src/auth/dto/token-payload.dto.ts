export class TokenPayloadDto {
  sub: string;
  username: string;
  iat: number;
  exp: number;
  aud: string;
  iss: string;
}
