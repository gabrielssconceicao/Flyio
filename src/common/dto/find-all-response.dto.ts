import { ApiProperty } from '@nestjs/swagger';

export class FindAllResponseDto<T> {
  @ApiProperty({
    example: 1,
    description: 'Total count of items in the database',
  })
  count: number;

  @ApiProperty({
    description: 'List of items',
    isArray: true,
  })
  items: T[];
}
