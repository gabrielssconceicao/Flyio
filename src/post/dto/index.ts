import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
