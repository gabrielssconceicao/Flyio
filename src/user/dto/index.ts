import { CreateUserDto } from './create-user.dto';
import { FindAllUsersResponseDto } from './find-all-users.dto';
import { QueryParamDto } from './query-param.dto';
import { ReactivateUserDto } from './reactivate-user.dto';
import { UpdateUserDto } from './update-user.dto';

export {
  CreateUserDto,
  FindAllUsersResponseDto,
  QueryParamDto,
  ReactivateUserDto,
  UpdateUserDto,
};
