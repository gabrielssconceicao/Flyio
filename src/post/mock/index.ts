import { postMock } from './post.mock';
import { generateCreatePostDtoMock } from './create-post.dto.mock';
export { postMock, generateCreatePostDtoMock };
