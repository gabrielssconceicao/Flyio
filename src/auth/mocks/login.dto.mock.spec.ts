import { generateLoginDtoMock } from './login.dto.mock';

describe('<LoginDtoMock>', () => {
  it('should generate a valid login dto mock', () => {
    const result = generateLoginDtoMock();
    const properties = ['login', 'password'];
    properties.forEach((property) => {
      expect(result).toHaveProperty(property);
    });
    expect(result).toEqual({
      login: 'login',
      password: '123456',
    });
    expect(result).toMatchSnapshot();
  });
});
