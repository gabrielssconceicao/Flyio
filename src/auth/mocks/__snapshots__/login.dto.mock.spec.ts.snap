// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginDtoMock> should generate a valid login dto mock 1`] = `
{
  "login": "login",
  "password": "123456",
}
`;
